### DISCUSSION TITLE
Mac support?

### DISCUSSION THREAD

**Discussion by classilla** [ProdigyReloaded/delivery-system#95] [General]
Haven't sat down with this yet, but how likely is this to work with the Mac client? I have a (prototype!) Mac Portable with the Prodigy client on it from its prior owner here. The version is "1.0 (C)1989".

While I'm on the subject, I was able to break down `STAGE.DAT` but the DOS `RS.EXE` pretty much hangs on everything inside it and I'm assuming endianness is likely to blame. The files seem slightly different than what you've got.

**Comment by pheller** [comment]
Hi there, apologies I didn't see this earlier.

The Prodigy Client for Mac Classic (1.0 as you describe) will definitely work with this implementation of the Delivery System and such a  configuration was exhibited at VCF West 2022.

Yes, as I recall, the STAGE.DAT for MacOS is the same, but the integers are stored as big endian.

See https://github.com/ProdigyReloaded/stageutil for a tool to extract the contents of that STAGE.DAT.  I would be very interested to see its contents since it sounds as if it was used in the past!

Please join us in the discord if you're not already there.  https://discord.gg/2GpsEC5JdP

**Comment by classilla** [comment]
I'm not sure it was used all that much, actually. However, I gave up working on it and deleted the files off my laptop, so I'll have to go dig out the Portable and fetch it from that. That might be a few days. I did try but I didn't get anything meaningful that anything else would process (though it's entirely possible and even probable I did something wrong).

**Reply by pheller** [reply]
No hurry and appreciate the help when you have an opportunity.

**Comment by classilla** [comment]
OK, here they are. I did try stageutil and I didn't get anything useful out of this, but I'm fully open to the possibility I screwed up somehow.
[mac-prodigy.zip](https://github.com/user-attachments/files/15807255/mac-prodigy.zip)

**Reply by pheller** [reply]
Ok, `CACHE.DAT` is empty.  I've extracted all the objects from `STAGE.DAT` and at a quick glance, it appears to just contain the objects that were likely distributed with in the installation itself.  That said, there may be a few more EAAsy Sabre objects in this `STAGE.DAT` than I recall seeing before.  Thanks for the sharing regardless!

**Reply by classilla** [reply]
How did you break them down and view them (what were the commands?)?

**Reply by pheller** [reply]
For `CACHE.DAT`, you can inspect with a hex editor and you will simply see NULL bytes.

For `STAGE.DAT`, you need to clone and build `stageutil`, which requires you have elixir installed (which in turn requires erlang).  The bits that follow are roughly what I do on MacOS.

```
git clone https://github.com/ProdigyReloaded/stageutil
cd stageutil
mix deps.get
mix escript.build
```

Then, run stageutil with the proper commands:

```
./stageutil export -r -d path/to/output path/to/stage.dat
```

You'll then find all the objects exported to the "output" directory, and you can view them with `CREDO`.
Find `CREDO` [here](https://archive.org/download/prodigy_202103/CREDO-2.2.46-03051991.IMG).

I run `CREDO` in DosBox-X.  Basically create a directory to be the DosBox "C" drive, make a Credo directory within and copy the files from the disk image linked above.
You'll then "mount" the path to the output objects in DosBox; something like "mount D: /path/to/output".
Run Credo, "Read" files in "D:\*.*", select something that will have some presentation data (files ending in `.MAP`, `.MEN`, `.PG`, `.BDY`, though some may have unresolvable references, so try a few).
Then, once an object is read, select "View Object" from the `CREDO` menu, and then ALT-F2 to display the object.

**Reply by classilla** [reply]
Fascinating! Thanks for the pointers. Yes, doesn't look all that exciting ... oh well. But the AppleLink install had some fun bits to explore.
